### PR TITLE
Update document-register-element dependency

### PR DESCRIPTION
**Pull request by erwinmombay**

### CODE REVIEW

File: test/size.txt [hunk 1]
```diff
@@ -1,61 +1,61 @@
...
       ---   |         ---   |         ---   |   ---
  72.23 kB   |     5.87 kB   |     2.53 kB   |   alp.js / alp.max.js
-817.55 kB   |   155.49 kB   |    48.72 kB   |   shadow-v0.js / amp-shadow.js
+835.71 kB   |   161.92 kB   |    51.16 kB   |   shadow-v0.js / amp-shadow.js
    5.8 kB   |       428 B   |       278 B   |   sw-kill.js / sw-kill.max.js
-337.38 kB   |       681 B   |       450 B   |   sw.js / sw.max.js
-852.21 kB   |   156.78 kB   |    49.21 kB   |   v0.js / amp.js
-293.45 kB   |    37.28 kB   |    12.58 kB   |   v0/amp-access-0.1.js
+355.54 kB   |       681 B   |       450 B   |   sw.js / sw.max.js
+870.37 kB   |   163.22 kB   |    51.66 kB   |   v0.js / amp.js
```
Comment (erwinmombay) [inline test/size.txt]: cc @cramforce if this is acceptable

Comment (cramforce) [inline test/size.txt]: I suppose this give us native shadow DOM/custom elements v1?

Comment (erwinmombay) [inline test/size.txt]: it gives us custom elements v1 correct as well as the install function dima needs to do shadow dom polyfilling but we can also do the latter with the old version by wrapping the script with a function ourselves.

**Comment by erwinmombay** [comment]
@dvoytenko @cramforce ready for review

File: build-system/tasks/compile.js [hunk 2]
```diff
@@ -310,18 +310,28 @@ function compile(entryModuleFilenames, outputDir,
...
+        'document-register-element.node.js').toString();
+    // CommonJS support for closure does not wrap the module
+    // and inject `global` like browserify and other node module loaders
+    // so we nee to change this to `self` or `window`.
+    file = file.replace('installCustomElements(global);',
+        'installCustomElements(self);');
+    // Closure Compiler does not generate a `default` property even though
+    // to interop CommonJS and ES6 modules. This is the same issue typescript
+    // ran into here https://github.com/Microsoft/TypeScript/issues/2719
+    file = file.replace('module.exports = installCustomElements;',
```
Comment (erwinmombay) [inline build-system/tasks/compile.js]: @cramforce PTAL at this

File: build-system/tasks/compile.js [hunk 3]
```diff
@@ -310,18 +310,28 @@ function compile(entryModuleFilenames, outputDir,
...
+        'exports.default = installCustomElements');
```
Comment (jridgewell) [inline build-system/tasks/compile.js]: You're missing the `;`.

File: build-system/tasks/compile.js [hunk 4]
```diff
@@ -310,18 +310,28 @@ function compile(entryModuleFilenames, outputDir,
...
-              'document-register-element.max.js') +
-          '\n\nexport function deadCode() {}\n');
-    }
+    file = fs.readFileSync(
+        'node_modules/document-register-element/build/' +
+        'document-register-element.node.js').toString();
+    // CommonJS support for closure does not wrap the module
+    // and inject `global` like browserify and other node module loaders
+    // so we nee to change this to `self` or `window`.
+    file = file.replace('installCustomElements(global);',
```
Comment (dvoytenko) [inline build-system/tasks/compile.js]: Why not just only have exported function and call it directly from `polyfill.js`?

File: build-system/tasks/compile.js (hunk 2 above)
Comment (cramforce) [inline build-system/tasks/compile.js]: If it works this is fine.

File: build-system/tasks/compile.js (hunk 4 above)
Comment (erwinmombay) [inline build-system/tasks/compile.js]: this was exactly the original code i wrote, unfortunately that would mean our babel/dev installation would be different from the closure compiler version and didn't want to make that split

File: build-system/tasks/compile.js (hunk 3 above)
Comment (erwinmombay) [inline build-system/tasks/compile.js]: good catch. added. thanks

File: build-system/tasks/compile.js (hunk 4 above)
Comment (dvoytenko) [inline build-system/tasks/compile.js]: ok

**Pull request merged**